databroker/core.py

```diff
@@ -1308,54 +1296,34 @@ def dataarray_page_to_dataset_page(dataarray_page):
             'filled': xarray.merge(dataarray_page['filled'].values())}
 
 
-class DaskFiller(event_model.Filler):
-
-    def __init__(self, *args, inplace=False, **kwargs):
-        if inplace:
-            raise NotImplementedError("DaskFiller inplace is not supported.")
-        # The DaskFiller will not mutate documents pass in by the user, but it
-        # will ask the base class to mutate *internal* copies in place, so we
-        # set inplace=True here, even though the user documents will never be
-        # modified in place.
-        super().__init__(*args, inplace=True, **kwargs)
-
-    def event_page(self, doc):
-
-        @dask.delayed
-        def delayed_fill(event_page, key):
-            self.fill_event_page(event_page, include=key)
-            return numpy.asarray(event_page['data'][key])
+def coerce_dask(handler_class, filler_state):
+    # If the handler has its own delayed logic, defer to that.
+    if hasattr(handler_class, 'return_type'):
+        if handler_class.return_type['delayed']:
```

Comment:
```suggestion
        if handler_class.return_type.get('delayed', False):
```

Reply:
The strict lookup was intentional. Do we want to go down the path of "If the `return_type` attribute exists, it can have whatever inside?" That level of flexibility has made the hints code very messy. I would rather say, "If `return_type` exists, it must take a position on 'delayed'."